model/map/src/main/java/org/keycloak/models/map/common/AbstractMapProviderFactory.java

```diff
@@ -78,4 +78,8 @@ public void init(Scope config) {
     public boolean isSupported() {
         return Profile.isFeatureEnabled(Profile.Feature.MAP_STORAGE);
     }
+
+    public Scope getStorageConfigScope() {
+        return storageConfigScope;
+    }
```

Comment:
I'm not sure this is the correct `scope` we would want to use. This only contains the `storage` config that is found inside the particular areas (`client`, `group`, etc). The final storage factory that is instantiated actually gets a `scope` that is broader than this - it combines the this area-specific config with the root config found in the `mapStorage` configuration (i.e. it falls back to the root config if a property is not defined in the area config). Also, if we go forward with multiple `profile`s for a particular storage, that would have to be taken into consideration as well.

So wouldn't we want to use the actual scope that was used to init the storage factory instead?

Reply:
@sguilhen @vramik This scope is the right one for the storage config, the options inherited from the root (and perhaps profile) are available transparently from this scope. See [`ComponentModelScope`](https://github.com/keycloak/keycloak/blob/b34f46155c494bb3944913eb94cd6bdb845eae33/server-spi-private/src/main/java/org/keycloak/component/ComponentModelScope.java#L67-L71). However, storage maintains the capability of the storage to respect letter case. Whether the lettercase should be respected _if the store supports it_ is however setting on the realm, so it has to be a property of the map user provider.

---

@vramik Factory should translate the configuration `Scope` values into properties and the objects created by the factory should not be given access to the `Scope` instance unless justified.

The reason is to keep the configuration management maintained in a single place (factory), and translated once (in the factory) rather than multiple times (in each created instance).

It seems that the only reason for this method is the eventually be searched for `username-lettercase` option. This by itself is not sufficient justification. Is there any other?